File: 2589. Minimum Time to Complete All Tasks/Solution.py
class Solution:
    def findMinimumTime(self, tasks: List[List[int]]) -> int:
        tasks = sorted(tasks, key = itemgetter(1))
        on = [False for i in range(2001)]
        count = 0
        for x, y, z in tasks:
            curr = 0
            for i in range(x, y + 1):
                if on[i]: curr += 1
            for i in range(y, x - 1, -1):
                if curr >= z: break
                if on[i]: continue
                on[i] = True
                count += 1
                curr += 1
        return count

File: 2589. Minimum Time to Complete All Tasks/Solution.java
class Solution {
    public int findMinimumTime(int[][] tasks) {
        Arrays.sort(tasks, (a, b) -> a[1] - b[1]);
        int count = 0;
        boolean[] on = new boolean[2001];
        for (int[] t : tasks) {
            int start = t[0], end = t[1], time = t[2], curr = 0;
            for (int i = start; i <= end; i++) {
                if (on[i]) curr++;
            }
            for (int i = end; curr < time && i >= start; i--) {
                if (on[i]) continue;
                on[i] = true;
                count++;
                curr++;
            }
        }
        return count;
    }
}